### PR TITLE
Update cssjanus 1.2 to 2.1, stop relying on PrestaShop fork

### LINKED ISSUE #25845: Use upstream CssJanus instead of fork
In 2018, PrestaShop forked https://github.com/cssjanus/php-cssjanus on https://github.com/PrestaShop/php-cssjanus .

This was because of an issue related to https://github.com/cssjanus/php-cssjanus/issues/14. Details on @eternoendless post: https://github.com/cssjanus/php-cssjanus/pull/13#issuecomment-355574605

This bug has been solved in latest php-cssjanus versions. We should be able to stop relying on our own outdated fork.

### PR DESCRIPTION
**Pull request by matks**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | See below
| Type?             | improvement
| Category?         | CO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/25845
| How to test?      | CI should be valid
| Possible impacts? | Only PrestaShop\PrestaShop\Core\Localization\RTL\StylesheetGenerator could be modified

## Description

In 2018, PrestaShop forked https://github.com/cssjanus/php-cssjanus on https://github.com/PrestaShop/php-cssjanus .

This was because of an issue related to https://github.com/cssjanus/php-cssjanus/issues/14. Details on @eternoendless post: https://github.com/cssjanus/php-cssjanus/pull/13#issuecomment-355574605

This bug has been solved in latest php-cssjanus versions.

This PR has 2 steps, 2 commits:

1. The first commit adds a unit test for the PrestaShop class StylesheetGenerator. This is the only class that relies on CSSJanus, and it uses CSSJanus to convert CSS from LTR to RTL.

The unit test fetches a CSS sample file `rtl_input.css` that I built using CSSJanus test samples, and runs `StylesheetGenerator::generateInDirectory()` to generate a `rtl_input_rtl.css` that is the RTL version of the input.

2. The second commits stops relying on our fork, and update CSSJanus from 1.2.0 to 2.1.0 .

The unit test added above verifies that the behavior of StylesheetGenerator is equivalent to before.

## BC Break

CSSJanus Composer dependency version update: from 1.2 to 2.1

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25847)

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]

File: tests/Unit/Core/Localization/RTL/StylesheetGeneratorTest.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    protected $sandboxDirectory;
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function setUp(): void
+    {
+        $this->filesystem = new Filesystem();
+        $this->sandboxDirectory = sys_get_temp_dir() . '/StylesheetGeneratorTest';
+        $this->cssSamplesDirectory = __DIR__ . '/../../../Resources/assets/css';
```
Comment (Progi1984) [inline tests/Unit/Core/Localization/RTL/StylesheetGeneratorTest.php]:
```suggestion
        $this->cssSamplesDirectory = dirname(__DIR__, 3) . '/Resources/assets/css';
```

File: tests/Unit/Core/Localization/RTL/StylesheetGeneratorTest.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        );
+    }
+
+    protected function tearDown(): void
+    {
+        $this->filesystem->remove($this->sandboxDirectory . '/' . self::RTL_INPUT_FILENAME);
+        $this->filesystem->remove($this->sandboxDirectory . '/' . self::RTL_INPUT_FILENAME_WITH_SUFFIX);
+    }
+
+    public function test()
```
Comment (Progi1984) [inline tests/Unit/Core/Localization/RTL/StylesheetGeneratorTest.php]:
```suggestion
    public function testGeneration(): void
```

**Comment by Progi1984** [comment]
Small comment

**Comment by matks** [comment]
> Small comment

You literally wrote a small comment 😝 

Capture d’écran 2021-09-11 à 13 32 13

**Review by Progi1984 — COMMENTED** [review]
@matks Thank you for your contribution. I have commented some improvements

**Comment by matks** [comment]
> @matks Thank you for your contribution. I have commented some improvements

I saw you 👀 

Capture d’écran 2021-09-13 à 10 47 50

I can see who you prefer, me or @PierreRambaud

File: tests/Unit/Core/Localization/RTL/StylesheetGeneratorTest.php (hunk 1 above)
Comment (matks) [reply tests/Unit/Core/Localization/RTL/StylesheetGeneratorTest.php]: I did not know about the second parameter of `dirname` so I learnt something new today! Thank you!

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thank you @matks

**Pull request merged**